module/zfs/dnode.c

```diff
@@ -1621,7 +1621,9 @@ dnode_rele_and_unlock(dnode_t *dn, void *tag, boolean_t evicting)
 	 * other direct or indirect hold on the dnode must first drop the dnode
 	 * handle.
 	 */
+#ifdef ZFS_DEBUG
 	ASSERT(refs > 0 || dnh->dnh_zrlock.zr_owner != curthread);
+#endif
```

Comment:
Why was this needed? Are the new ASSERT definitions not happy with this when !ZFS_DEBUG?

Reply:
Yes, `zr_owner` only exists when ZFS_DEBUG. This is the only instance of the new asserts requiring this, so I decided against making an ASSERT_SWALLOW.